util/codec/codec.go

```diff
@@ -759,6 +759,52 @@ func DecodeRange(b []byte, size int) ([]types.Datum, []byte, error) {
 	return values, nil, nil
 }
 
+// DecodeIndexRange decodes the range values from a byte slice that generated by EncodeKey.
+// It handles some special values like `MinNotNull` and `MaxValueDatum`.
+func DecodeIndexRange(b []byte, size int, idxColumnType []byte) ([]types.Datum, []byte, error) {
```

Comment:
ditto, I advise adding the new argument to the original function instead of creating a new one since they are similar.